# Profitable App Profiles for the App Store and Google Play Markets

In this project, our goal is to find mobile app profiles that are profitable for the App Store and the google play markets. We will be working for a company that creates mobile apps compatible with IOS and Android. We will be working as data analysts in this project and our goal is to enable our team of developers to make data-driven decisions with respect to the kind of apps they build.

At our company we only create free apps and our main source of revenue comes from in-app ads. Meaning our revenue for a a given app will be influenced by number of people using our app. Our goal for this project is to analyze data to help our developers understand what kinds of apps are likely to attract more users.


# Opening and Exploring Data

In 2018 it was seen that the total number of apps available on the IOS app store are approximately 2 million. the total number of apps available on the Google PlayStore was approximately 2.1 million. 

It would be very tedious to collecct information for over 4 million apps. However there are two data sets available for our usage.

The data set for Google PlayStore can be found [here](https://www.kaggle.com/lava18/google-play-store-apps).
The data set for IOS App store can be found [here](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps)

We start by opening the data sets.



In [1]:
from csv import reader

#Google Playstore
opened_file = open('googleplaystore.csv')
read_file = reader(opened_file)
playstore = list(read_file)
playstore_header = playstore[0]
playstore_data = playstore[1:]

#App Store IOS
opened_file = open('AppleStore.csv')
read_file = reader(opened_file)
appstore = list(read_file)
appstore_header = appstore[0]
appstore_data = appstore[1:]

Now that we have opened our data sets, we create a function called explore_data which displays the data for us and lets us know the number of columns and rows in each data set.

In [2]:
def explore_data(dataset, start, end, rows_and_columns=False):
    dataset_slice = dataset[start:end]    
    for row in dataset_slice:
        print(row)
        print('\n') 

    if rows_and_columns:
        print('Number of rows:', len(dataset))
        print('Number of columns:', len(dataset[0]))
        
print(playstore_header)
print('\n')
explore_data(playstore_data,0,4,True)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 10841
Number of columns: 13


After using the explore_data( ) function on the google Play Store data set, we see that the total number of rows are 10841 and the total number of columns are 13. The columns that seem interesting/relevant to us in regards to our project are: 'App', 'Category', 'Reviews', 'Installs', 'Type', 'Price' and 'Genres'.

In [3]:
print(appstore_header)
print('\n')
explore_data(appstore_data, 0, 4, True)

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


['420009108', 'Temple Run', '65921024', 'USD', '0.0', '1724546', '3842', '4.5', '4.0', '1.6.2', '9+', 'Games', '40', '5', '1', '1']


Number of rows: 7197
Number of columns: 16


After using the explore_data( ) function on the data set for the IOS appstore, we see that the total number of rows are 7197 and total number of columns are 16. The columns that seem interesting in this data set are: 'track_name', 'currency', 'price', 'rating_count_tot', 'rating_count_ver' and 'prime_genre'.

Not all the given column names are self explanatory in the case of the IOS data set, please refer to this  [documentation](https://www.kaggle.com/lava18/google-play-store-apps) for more details.

# Data Cleaning

In the [dedicated discussion section](https://www.kaggle.com/lava18/google-play-store-apps/discussion) for the Google PlayStore data, we see that there is an error outlined for row 10472. We print his row next to our header row to find out the error.

In [4]:
print(playstore_data[10472])
print('\n')
print(playstore_header)

['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']


['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


We see that row 10472 corresponds to an app named 'Life Made WI-Fi Touchscreen Photo Frame'. The error in this row is that the rating for this app is 19 which is not possible since the rating cannot be above 5. This error is caused because there is a missing entry in the 'Category' column of this row. Hence we delete this row.

In [5]:
del playstore_data[10472]
print(len(playstore_data))

10840


# Duplicate Data Deletion

Exploring the google PlayStore data set, we find that there exists apps that have multiple entries in the data set. Also, If we check the [discussion section](https://www.kaggle.com/lava18/google-play-store-apps/discussion) of the Google PlayStore data set, we see that multiple entries are indeed a problem. 

Lets check to see how many duplicate apps there are in total. 

In [6]:
duplicate_apps = []
unique_apps = []

for app in playstore_data:
    name = app[0]
    if name in unique_apps:
        duplicate_apps.append(name)
    else:
        unique_apps.append(name)
        
print('number of duplicate apps:', len(duplicate_apps))
print('\n')
print('Examples of duplicate apps:', duplicate_apps[:10])

number of duplicate apps: 1181


Examples of duplicate apps: ['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings', 'Box', 'Zenefits', 'Google Ads', 'Google My Business', 'Slack']


Since we dont want apps to be counted more than once, we will dlete the multiple app entries and keep only one.

Lets examine an example of an app with multiple entries. We will take instagram for example. 

In [7]:
for app in playstore_data:
    name = app[0]
    if name == 'Instagram':
        print(app)

['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577446', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66509917', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']


After examining the rows, we see that the difference takes place in the fourth column of each row which correlates to the number of reviews for that app. The different numbers show us that the data was collected at different times. We can use this information to make our own critereon. So which row do we keep? We will keep the rows with the highest number of reviews. The higher the reviews , the more recent and more reliable the data. How do we do that?

- Create a dictionary where each key is a unique app name, and the value is the highest number of reviews of that app
- Use the information stored in the dictionary and create a new data set, which will have only one entry per app (and for each app, we'll only select the entry with the highest number of reviews).

In [8]:
reviews_max = {}

for app in playstore_data:
    name = app[0]
    n_reviews = float(app[3])
    
    if name in reviews_max and reviews_max[name] < n_reviews:
        reviews_max[name] = n_reviews
        
    elif name not in reviews_max:
        reviews_max[name] = n_reviews
        
        
print('Expected Length:', len(playstore_data)-1181)
print('Actual Lenght:', len(reviews_max))

Expected Length: 9659
Actual Lenght: 9659


In the code cell below, we use the __reviews_max__ dictionary we created earlier to remove duplicate rows. We start this process by creating two empty lists called __android_clean__(which will contain our new clean data) and __already_added__(which will store app names).

Once the Lists are created, we loop through the __playstore_data__ and for every iteration, we assign the name of the app to a variable called __name__ and we convert the number of reviews for every app to a float and store it in a variable called __n_reviews__.

Next we use an if statement inside the body of the for loop. if __n_reviews__ is the same as the number of maximum reviews of the app __name__ _and_ name is not already in the list __already_added__:
- Append the row to the list __android_clean__.
- Append the name of the app __name__ to the list __already_added__.

Finally we check to see if the length of the list __android_clean__ is the same as we expected by using the explore function. 

In [9]:
android_clean = []
already_added = []

for app in playstore_data:
    name = app[0]
    n_reviews = float(app[3])
    
    if (reviews_max[name] == n_reviews) and (name not in already_added):
        android_clean.append(app)
        already_added.append(name)
        
explore_data(android_clean, 0, 3, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 9659
Number of columns: 13


# Deleting Non-English Apps

If we analyze both of the data sets, we see that thereare apps that are directed towards non english users however we are only interested in apps that are directed towards english speaking users. Our next step is to sort the data and remove all the non-English apps.

First we will create a function that takes in the app name as a string and tells us if the app is directed toward english users or non english users.

Every character has a number associated with it. The numbers corresponding to characters that we use mostly in english text are all in the rane from 0 to 127 according to the ASCII. Also, there are many apps for English users that have the other special symbols in their name like emojis, em dash, en dash, etc. The number associated with these characters do not fall in the range of 0 to 127 for english characters in the ASCII guidelines. 

We will create a function that takes in the name of an app as a string and checks each character of that string. If there are more than 3 characters in the app name that are non Enlgish character, the function will return false which means the app is not suitable for English users. Once the function is created, we will test it out with a few app names each containing differnt types of symbols and characters.

In [10]:
def english(string):
    not_ascii = 0
    
    for character in string:
        if ord(character) > 127:
            not_ascii += 1
            
    if not_ascii > 3:
        return False
    else:
        return True
    
print(english('Instagram'))
print(english('爱奇艺PPS -《欢乐颂2》电视剧热播'))
print(english('Docs To Go™ Free Office Suite'))
print(english('Instachat 😜'))
    

True
False
True
True


Now that we have created and tested the funtion to filter out the english apps, we will use it for both the IOS and Google data sets. Once we have filtered the apps, we will explore a few rows of the datasets using the explore_data( ) function we created earlier.

In [11]:
android_eng = []
ios_eng = []

for app in android_clean:
    name = app[0]
    if english(name):
        android_eng.append(app)
        
for app in appstore_data:
    name = app[1]
    if english(name):
        ios_eng.append(app)
        
explore_data(android_eng, 0, 3, True)
print('\n')
explore_data(ios_eng, 0, 3, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 9614
Number of columns: 13


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 

# Isolating the Free Apps

Since our company works only with free apps and gains its revenue from in-app ads, we will also have to filter out the apps that are not free from our data sets. after isolating the free apps, we will have our final data sets since this is the last stage of the data cleaning process.

In [12]:
android_final = []
ios_final = []

for app in android_eng:
    price = app[7]
    if price == '0':
        android_final.append(app)
        
for app in ios_eng:
    price = app[4]
    if price == '0.0':
        ios_final.append(app)
        
print('number of android apps: ',len(android_final))
print('number of ios apps: ',len(ios_final))

number of android apps:  8864
number of ios apps:  3222


# Most Common Apps by Genre

In order to reach our goal determining the kinds of apps that are likely to attract more users, we also must find the app profiles that are successful on both markets. For example, an app profile that might work well with both markets would be some sort of productivity app that makes use of gamification.

We begin the analysis by finding out what the most common genres are for each of the two markets. To do this, we build frequency tables for the two app markets which give us the genre of
each app.

We create two functions that can 


In [13]:
def freq_table(dataset, index):
    table = {}
    total = 0
    
    for row in dataset:
        total += 1
        value = row[index]
        if value in table:
            table[value] += 1
        else:
            table[value] = 1
    
    table_percentages = {}
    for key in table:
        percentage = (table[key] / total) * 100
        table_percentages[key] = percentage 
    
    return table_percentages


def display_table(dataset, index):
    table = freq_table(dataset, index)
    table_display = []
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)
        
    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1], ':', entry[0])

Now we examine the column named prime genre of our ios store data set by using our frequency table function.

In [14]:
display_table(ios_final , -5)

Games : 58.16263190564867
Entertainment : 7.883302296710118
Photo & Video : 4.9658597144630665
Education : 3.662321539416512
Social Networking : 3.2898820608317814
Shopping : 2.60707635009311
Utilities : 2.5139664804469275
Sports : 2.1415270018621975
Music : 2.0484171322160147
Health & Fitness : 2.0173805090006205
Productivity : 1.7380509000620732
Lifestyle : 1.5828677839851024
News : 1.3345747982619491
Travel : 1.2414649286157666
Finance : 1.1173184357541899
Weather : 0.8690254500310366
Food & Drink : 0.8069522036002483
Reference : 0.5586592178770949
Business : 0.5276225946617008
Book : 0.4345127250155183
Navigation : 0.186219739292365
Medical : 0.186219739292365
Catalogs : 0.12414649286157665


We see that the App Store is dominated by apps that are designed fun (games, photo and video, entertainment, social networking) while apps with a more practical purpose are more rare.

now we move onto the google play store data set and we examine the 'category' and 'genre' columns.

In [15]:
display_table(android_final, 1) # Category

FAMILY : 18.907942238267147
GAME : 9.724729241877256
TOOLS : 8.461191335740072
BUSINESS : 4.591606498194946
LIFESTYLE : 3.9034296028880866
PRODUCTIVITY : 3.892148014440433
FINANCE : 3.7003610108303246
MEDICAL : 3.531137184115524
SPORTS : 3.395758122743682
PERSONALIZATION : 3.3167870036101084
COMMUNICATION : 3.2378158844765346
HEALTH_AND_FITNESS : 3.0798736462093865
PHOTOGRAPHY : 2.944494584837545
NEWS_AND_MAGAZINES : 2.7978339350180503
SOCIAL : 2.6624548736462095
TRAVEL_AND_LOCAL : 2.33528880866426
SHOPPING : 2.2450361010830324
BOOKS_AND_REFERENCE : 2.1435018050541514
DATING : 1.861462093862816
VIDEO_PLAYERS : 1.7937725631768955
MAPS_AND_NAVIGATION : 1.3989169675090252
FOOD_AND_DRINK : 1.2409747292418771
EDUCATION : 1.1620036101083033
ENTERTAINMENT : 0.9589350180505415
LIBRARIES_AND_DEMO : 0.9363718411552346
AUTO_AND_VEHICLES : 0.9250902527075812
HOUSE_AND_HOME : 0.8235559566787004
WEATHER : 0.8009927797833934
EVENTS : 0.7107400722021661
PARENTING : 0.6543321299638989
ART_AND_DESIGN : 

We see that unlike the App Store, the Google Playstorehas a much different landscape. There are not as many apps designed for fun. However, there are a good number of apps that are designed for practical use. 

In [16]:
display_table(android_final, -4) #genre

Tools : 8.449909747292418
Entertainment : 6.069494584837545
Education : 5.347472924187725
Business : 4.591606498194946
Productivity : 3.892148014440433
Lifestyle : 3.892148014440433
Finance : 3.7003610108303246
Medical : 3.531137184115524
Sports : 3.463447653429603
Personalization : 3.3167870036101084
Communication : 3.2378158844765346
Action : 3.1024368231046933
Health & Fitness : 3.0798736462093865
Photography : 2.944494584837545
News & Magazines : 2.7978339350180503
Social : 2.6624548736462095
Travel & Local : 2.3240072202166067
Shopping : 2.2450361010830324
Books & Reference : 2.1435018050541514
Simulation : 2.0419675090252705
Dating : 1.861462093862816
Arcade : 1.8501805054151623
Video Players & Editors : 1.7712093862815883
Casual : 1.7599277978339352
Maps & Navigation : 1.3989169675090252
Food & Drink : 1.2409747292418771
Puzzle : 1.128158844765343
Racing : 0.9927797833935018
Role Playing : 0.9363718411552346
Libraries & Demo : 0.9363718411552346
Auto & Vehicles : 0.9250902527075

When we compare the category and genre columns for the google play stoer, we see that the genre columns has many more categories. Since we are looking for the bigger picuture for now, we will disregard the genre columns and work with the category column only. 

We've found that the google playstore houses a balanced landscape  of both practical and for-fun apps, while the App store has majority of apps designed for fun. 

## Most Popular Apps by Genre on the App Store

We can find out what genres are the most popular by looking at the average number of installs for each of the app genres. since the apple App Store does not have such a column in its data set, we look at the *rating_count_tot* column instead which gives us the number of user ratings.

We calculate the average number of user ratings per app genre on the App Store:

In [17]:
genres_ios = freq_table(ios_final, -5)

for genre in genres_ios:
    total = 0
    len_genre = 0
    for app in ios_final:
        genre_app = app[-5]
        if genre_app == genre:
            n_ratings = float(app[5])
            total += n_ratings
            len_genre += 1
    avg_n_ratings = total / len_genre
    print(genre, ':', avg_n_ratings)

Social Networking : 71548.34905660378
Photo & Video : 28441.54375
Games : 22788.6696905016
Music : 57326.530303030304
Reference : 74942.11111111111
Health & Fitness : 23298.015384615384
Weather : 52279.892857142855
Utilities : 18684.456790123455
Travel : 28243.8
Shopping : 26919.690476190477
News : 21248.023255813954
Navigation : 86090.33333333333
Lifestyle : 16485.764705882353
Entertainment : 14029.830708661417
Food & Drink : 33333.92307692308
Sports : 23008.898550724636
Book : 39758.5
Finance : 31467.944444444445
Education : 7003.983050847458
Productivity : 21028.410714285714
Business : 7491.117647058823
Catalogs : 4004.0
Medical : 612.0


We see that the Navigtion apps have the highest number of user reviews on average. But this is because it is heavily infuenced by Waze and Google maps, which have almost half a million reviews together. 

So we find the name and number of ratings for the navigation apps.

In [18]:
for app in ios_final:
    if app[-5] == 'Navigation':
        print(app[1], ':', app[5]) 

Waze - GPS Navigation, Maps & Real-time Traffic : 345046
Google Maps - Navigation & Transit : 154911
Geocaching® : 12811
CoPilot GPS – Car Navigation & Offline Maps : 3582
ImmobilienScout24: Real Estate Search in Germany : 187
Railway Route Search : 5


We can see that this pattern repeats itself in other genres of apps as well. For example, social networking apps are heavily influenced by huge apps like Facebook, Pinterest, Skype etc. Or in the caseof music apps where Spotify, Shazam or Pandora influence the avergae number.

We see that reference apps also have a high number of user ratingss. 

In [19]:
for app in ios_final:
    if app[-5] == 'Reference':
        print(app[1], ':', app[5])

Bible : 985920
Dictionary.com Dictionary & Thesaurus : 200047
Dictionary.com Dictionary & Thesaurus for iPad : 54175
Google Translate : 26786
Muslim Pro: Ramadan 2017 Prayer Times, Azan, Quran : 18418
New Furniture Mods - Pocket Wiki & Game Tools for Minecraft PC Edition : 17588
Merriam-Webster Dictionary : 16849
Night Sky : 12122
City Maps for Minecraft PE - The Best Maps for Minecraft Pocket Edition (MCPE) : 8535
LUCKY BLOCK MOD ™ for Minecraft PC Edition - The Best Pocket Wiki & Mods Installer Tools : 4693
GUNS MODS for Minecraft PC Edition - Mods Tools : 1497
Guides for Pokémon GO - Pokemon GO News and Cheats : 826
WWDC : 762
Horror Maps for Minecraft PE - Download The Scariest Maps for Minecraft Pocket Edition (MCPE) Free : 718
VPN Express : 14
Real Bike Traffic Rider Virtual Reality Glasses : 8
教えて!goo : 0
Jishokun-Japanese English Dictionary & Translator : 0


We see that the average ratings are skewed once again because of Bible and Dictionary.com. 

We notice that this niche seems to have potential. What we could possibly do is to take another book and turn it into an app and which includes more feateures than just the raw version of the book itself. This can include quotes from the book, quizzes, or even audio version of the book, etc. Moreover, we can even add a dictionary within this app so that the user does not have to exit our app and go to another app to search up words.

This seems o be a good idea in the case of the App Store since the majority of apps seem to be for-fun apps and a practical app like ours will have a greater chance to stand out among all the apps on the App Store.

Other app genres that are popular in the App Store include Weather, Book, Food and Drink, and Finance. Among these apps, the Book genre is the only one that seems to overlap with our idea, other than that, we are not interested in the other app genres.

## Most Popular Apps by Genre on Google Play

To find out the most popular apps by genre, we are given a column with the number of installs in the google play market data set. Lets have a a look at this column.

In [20]:
display_table(android_final, 5)

1,000,000+ : 15.726534296028879
100,000+ : 11.552346570397113
10,000,000+ : 10.548285198555957
10,000+ : 10.198555956678701
1,000+ : 8.393501805054152
100+ : 6.915613718411552
5,000,000+ : 6.825361010830325
500,000+ : 5.561823104693141
50,000+ : 4.7721119133574
5,000+ : 4.512635379061372
10+ : 3.5424187725631766
500+ : 3.2490974729241873
50,000,000+ : 2.3014440433213
100,000,000+ : 2.1322202166064983
50+ : 1.917870036101083
5+ : 0.78971119133574
1+ : 0.5076714801444043
500,000,000+ : 0.2707581227436823
1,000,000,000+ : 0.22563176895306858
0+ : 0.04512635379061372
0 : 0.01128158844765343


We see that the values given are open ended meaning the data is not really precise. we dont know whether an app with 100,000+ installs has 100,000, 200,000 or 350,000. 

Since we only want to get an idea of which types of apps attract the most users, we dont need very precise data.

We will leave the numbers as they are, which means that we will consider an app with 100,000+ installs has 100,000 installs, and an app with 1,000,000+ installs has 1,000,000 installs and so on.

To perform computations , we will convert each number to a float and we will remove the comma and plus characters so that we dont get an error. 

We will do this in a loop below, where we also compute the average number of installs for each genre.

In [21]:
categories_android = freq_table(android_final, 1)

for category in categories_android:
    total = 0
    len_category = 0
    for app in android_final:
        category_app = app[1]
        if category_app == category:
            n_installs = app[5]
            n_installs = n_installs.replace(',','')
            n_installs = n_installs.replace('+','')
            total += float(n_installs)
            len_category += 1
    avg_n_installs = total/len_category
    print(category, ':', avg_n_installs)

ART_AND_DESIGN : 1986335.0877192982
AUTO_AND_VEHICLES : 647317.8170731707
BEAUTY : 513151.88679245283
BOOKS_AND_REFERENCE : 8767811.894736841
BUSINESS : 1712290.1474201474
COMICS : 817657.2727272727
COMMUNICATION : 38456119.167247385
DATING : 854028.8303030303
EDUCATION : 1833495.145631068
ENTERTAINMENT : 11640705.88235294
EVENTS : 253542.22222222222
FINANCE : 1387692.475609756
FOOD_AND_DRINK : 1924897.7363636363
HEALTH_AND_FITNESS : 4188821.9853479853
HOUSE_AND_HOME : 1331540.5616438356
LIBRARIES_AND_DEMO : 638503.734939759
LIFESTYLE : 1437816.2687861272
GAME : 15588015.603248259
FAMILY : 3695641.8198090694
MEDICAL : 120550.61980830671
SOCIAL : 23253652.127118643
SHOPPING : 7036877.311557789
PHOTOGRAPHY : 17840110.40229885
SPORTS : 3638640.1428571427
TRAVEL_AND_LOCAL : 13984077.710144928
TOOLS : 10801391.298666667
PERSONALIZATION : 5201482.6122448975
PRODUCTIVITY : 16787331.344927534
PARENTING : 542603.6206896552
WEATHER : 5074486.197183099
VIDEO_PLAYERS : 24727872.452830188
NEWS_AND_

We see that on average, the communication apps have the most number of installs. which is 38,456,119. 

In [22]:
for app in android_final:
    if app[1] == 'COMMUNICATION' and (app[5]=='1,000,000,000+' 
                                     or app[5] == '500,0000,000+'
                                     or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

WhatsApp Messenger : 1,000,000,000+
imo beta free calls and text : 100,000,000+
Android Messages : 100,000,000+
Messenger – Text and Video Chat for Free : 1,000,000,000+
Skype - free IM & video calls : 1,000,000,000+
Who : 100,000,000+
GO SMS Pro - Messenger, Free Themes, Emoji : 100,000,000+
Google Chrome: Fast & Secure : 1,000,000,000+
Firefox Browser fast & private : 100,000,000+
Gmail : 1,000,000,000+
Hangouts : 1,000,000,000+
Messenger Lite: Free Calls & Messages : 100,000,000+
Kik : 100,000,000+
KakaoTalk: Free Calls & Text : 100,000,000+
Opera Mini - fast web browser : 100,000,000+
Opera Browser: Fast and Secure : 100,000,000+
Telegram : 100,000,000+
Truecaller: Caller ID, SMS spam blocking & Dialer : 100,000,000+
UC Browser Mini -Tiny Fast Private & Secure : 100,000,000+
WeChat : 100,000,000+
Yahoo Mail – Stay Organized : 100,000,000+
BBM - Free Calls & Messages : 100,000,000+


Just like the App Store we see tha this number is heavily skewed due to some giant apps. In this case it would be apps like (Whatsapp, Facebook Messenger, Skype, Google Chrome, Gmail and Hangouts)

If we remove all the communication apps that have over 100 million installs, the average would then be reduced about 10 times.

In [23]:
under_100_m = []

for app in android_final:
    n_installs = app[5]
    n_installs = n_installs.replace(',','')
    n_installs = n_installs.replace('+','')
    if (app[1] == 'COMMUNICATION') and (float(n_installs)< 100000000):
        under_100_m.append(float(n_installs))
        
sum(under_100_m)/len(under_100_m)

3603485.3884615386

Regardless of the average installs, we still know that the communications genre is very saturated because of of some giants that are dominating the niches. 

We look at the genre with the least amount of average installs which is 'COMICS'. We can choose to create an app that targets comic book readers. To make this app better, we can include all many genres of comics in this app for example superhero, manga, humour, non-fiction, etc. We can include a language translation feature in the app and also a audio version of the comics as well. This will make our app more complete and it will make it stand out more. 


## Conclusion 

In this project, we analyzed two data sets. One was about the Google Playstore and the other was the App Store from apple. our goal was to recommend app profile that will be successful after analyzing the two data sets.

We had different recommendations for each of the app markets. For the IOS App Store, we recommended an app that takes a popular book and turn it into an app and by adding features like a dictionary inside the app, audio version of the book, daily quotes, we make our app stand out more.

For the Google Playstore we recommended creating an app for comic book readers which includes many different genres of comics and also features like audio versions of comics language translation so that we have a bigger audience for our apps. 